reconcile.go

```diff
@@ -0,0 +1,264 @@
+package autopilot
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/raft"
+)
+
+// reconcile calculates and then applies promotions and demotions
+func (a *Autopilot) reconcile() error {
```

Comment:
Unless I missed something, I don't think we have any tests checking that we apply changes correctly in the `reconcile` loop?

Reply:
Yep, just added a bunch of tests for this now (and all of the reconcile.go funcs)